src/classes/coreData.cpp

```diff
@@ -528,6 +531,64 @@ Configuration *CoreData::findConfigurationByNiceName(std::string_view name) cons
     return it->get();
 }
 
+/*
+ * Layers
+ */
+
+// Add new processing layer
+ModuleLayer *CoreData::addProcessingLayer() { return processingLayers_.emplace_back(std::make_unique<ModuleLayer>()).get(); }
+
+// Remove processing layer
+void CoreData::removeProcessingLayer(ModuleLayer *layer)
+{
+    if (!layer)
+        return;
+
```

Comment:
Don't we need this in here?
```suggestion

    // Remove any references to the Modules in the layer before we delete it
    for (auto &module : layer->modules())
        removeReferencesTo(module.get());
        
```

Reply:
Yes, we probably do!